projects/igniteui-angular/src/lib/snackbar/snackbar.component.spec.ts

```diff
@@ -90,6 +91,64 @@ describe('IgxSnackbar', () => {
     });
 });
 
+describe('Custom content', () => {
+    configureTestSuite();
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                SnackbarCustomContentComponent
+            ],
+            imports: [
+                BrowserAnimationsModule,
+                IgxSnackbarModule
+            ]
+        }).compileComponents();
+    }));
+
+    let fixture, domSnackbar, snackbar;
+    beforeEach(async(() => {
+        fixture = TestBed.createComponent(SnackbarCustomContentComponent);
+        fixture.detectChanges();
+        snackbar = fixture.componentInstance.snackbar;
+        domSnackbar = fixture.debugElement.query(By.css('igx-snackbar')).nativeElement;
+    }));
+
+    it('should display a message', () => {
+        fixture.componentInstance.text = 'Undo';
+        snackbar.message = 'Item shown';
+        snackbar.isVisible = true;
+        fixture.detectChanges();
+
+        expect(domSnackbar.innerText).toContain('Item shown');
+
+        const messageEl = fixture.debugElement.query(By.css('.igx-snackbar__message'));
+        expect(messageEl).toBeTruthy('Message is not found');
+
+        const customContent = fixture.debugElement.query(By.css('.igx-snackbar__content'));
+        expect(customContent).toBeTruthy('Custom content is not found');
+
+        // Verify the message is displayed on the left side of the custom content
+        const messageElRect = (<HTMLElement>messageEl.nativeElement).getBoundingClientRect();
+        const customContentRect = (<HTMLElement>customContent.nativeElement).getBoundingClientRect();
+        expect(messageElRect.right <= customContentRect.left).toBe(true, 'The message is not on the left of the custom content');
+    });
+
+    it('should dispay custom content on the left side of the action button', () => {
+        fixture.componentInstance.text = 'Undo';
+        snackbar.isVisible = true;
+        fixture.detectChanges();
+
+        const customContent = fixture.debugElement.query(By.css('.igx-snackbar__content'));
+        expect(customContent).toBeTruthy('Custom content is not found');
+
+        // Verify the custom content element is on the left side of the button
+        const button = fixture.debugElement.query(By.css('.igx-snackbar__button'));
+        const buttonRect = (<HTMLElement>button.nativeElement).getBoundingClientRect();
+        const customContentRect = (<HTMLElement>customContent.nativeElement).getBoundingClientRect();
+        expect(customContentRect.right <= buttonRect.left).toBe(true, 'The custom element is not on the left of the button');
+    });
+});
+
```

Comment:
We should probably add a test with both a message and some content.

Reply:
The tests are updated so they check if the elements are positioned in the correct order when a message, a custom content and a button are set, and also that the message span is not in the DOM when the message is empty.